src/shared/components/QueryBuilder/domain/actions.ts

```diff
@@ -113,30 +105,29 @@ export const actions: any = {
   // eslint-disable-next-line sonarjs/cognitive-complexity
   editFilterOperator: assign((context: QueryBuilderContext, event: SelectEvent) => {
     if (context.edition === null) {
-      throw new Error('Cannot edit filter attribute without edition data!');
+      throw new Error('Cannot edit filter operator without edition data!');
     }
 
     const { filterId } = context.edition;
     const newOperator = event.data;
     let newEdition = null;
 
     const newFilters = context.filters.map((filter) => {
-      if (filter.id !== filterId) {
-        return filter;
-      }
-
       const previousOperator = filter.operator!.value;
 
-      if (previousOperator === OperatorKind['is-empty']) {
-        filter.value = { value: '', label: '' };
+      if (filter.id !== filterId || previousOperator === newOperator.value) {
+        return filter;
       }
 
       if (newOperator.value === OperatorKind['is-empty']) {
-        return {
+        return buildIsEmptyFilter({
           ...filter,
-          ...IsEmptyFilter,
           active: false,
-        };
+        });
+      }
+
+      if (previousOperator === OperatorKind['is-empty']) {
+        filter.value = { value: '(no value)', label: '(no value)' };
```

Comment:
we do this to give a hint to the user, as opposed to leave it empty and having some empty space in the UI